internal/backend/windows/win32_windows.go

```diff
@@ -228,14 +227,29 @@ func (Backend) WslUnregisterDistribution(distributionName string) (err error) {
 		return errors.New("could not convert distro name to UTF16")
 	}
 
-	r1, _, _ := apiWslUnregisterDistribution.Call(uintptr(unsafe.Pointer(distroUTF16)))
-
-	if r1 != 0 {
-		return fmt.Errorf("failed syscall")
+	_, err = callDll(apiWslUnregisterDistribution, uintptr(unsafe.Pointer(distroUTF16)))
+	if err != nil {
+		return err
 	}
+
 	return nil
 }
 
+func callDll(proc *syscall.LazyProc, args ...uintptr) (uint32, error) {
```

Comment:
Even though this is internal, it willl be helpful for future maintainers to know the reason why this wrapper function. Could you add a comment line for that purpose?